plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -973,9 +979,40 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
                 .orElseGet(DefaultAWSCredentialsProviderChain::getInstance);
 
         if (iamRole != null) {
+            String stsEndpointOverwrite = conf.get(S3_STS_ENDPOINT);
+            String stsRegionOverwrite = conf.get(S3_STS_REGION);
+
+            AWSSecurityTokenServiceClientBuilder stsClientBuilder =
+                    AWSSecurityTokenServiceClientBuilder.standard()
+                    .withCredentials(provider);
+
+            String region;
+            if (!isNullOrEmpty(stsRegionOverwrite)) {
+                region = stsRegionOverwrite;
+            } 
+            else {
+                DefaultAwsRegionProviderChain regionProviderChain =
+                        new DefaultAwsRegionProviderChain();
```

Comment:
No need to wrap this line
```suggestion
                DefaultAwsRegionProviderChain regionProviderChain = new DefaultAwsRegionProviderChain();
```